src/main/java/org/example/listener/InsultListener.java

```diff
@@ -0,0 +1,65 @@
+package org.example.listener;
+
+import net.dv8tion.jda.api.events.guild.GuildReadyEvent;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import org.example.manager.SlashCommandManager;
+import org.example.service.EventService;
+import org.example.service.InsultService;
+import org.example.slashcommand.SlashCommand;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+public class InsultListener extends ListenerAdapter {
```

Comment:
Why separate listener ? It's logic looks like a duplicate of GlobalListener and additional method is now needed in SlashCommandManager `org.example.manager.SlashCommandManager#getInsultSlashCommandsData`. If splitting listeners is really required, I would think of a way to avoid such code duplication (maybe using an abstract class).

Reply:
+ and addressed.
Listeners are separate because their responsibility is different.
They now inherit from parent abstract class "Listener" to reduce code duplication.